Automattic/wp-calypso
# Framework: Remove lib/preferences in favor of its Redux counterpart

#### Changes proposed in this Pull Request

* TinyMCE: Use Redux for AtD ignores
* State: Use Redux preferences in `saveEdited` posts action creator
~* Enable spellchecker for all site types - MUST BE DELETED BEFORE MERGING~
* Framework: Remove `lib/preferences` completely

#### Testing instructions

* Checkout this branch.
* Select a VIP site and open a post in the classic editor.
* Type some typo in the editor, "exxample" for example 😉 . 
* Click the "Proofread Writing" button from the toolbar.
* Verify that word gets underlined. 
* Click on it, verify you get suggestions.
* Click "ignore always" at the bottom and verify you never get suggestions to improve that word, even after a refresh.
* Save the post a couple of times and verify it works well and there are no console errors.

Note: the [third commit](https://github.com/Automattic/wp-calypso/commit/270060dcecbd54a401e6df99884105118c610e5b) must be deleted before merging - it was created just to enable the spellchecker on any site, which makes it easier for people who don't have a VIP site for testing.
